client/client_with_secret_test.go

```diff
@@ -215,3 +217,67 @@ func Test_GivenTokenSourceWithTokenThatWillNotExpireBetweenRequests_WhenTokenSto
 	mockedAuthService.AssertExpectations(t)
 	mockedClient.AssertExpectations(t)
 }
+
+func Test_GivenAServerThatRejectsHTTPCall_WhenAGetCallIsFired_ThenExpectTheClientToRenewAccessTokenOnce(t *testing.T) {
+	// Given
+	mockedAuthService := mocks.AuthService{}
+	mockedClient := mocks.Client{}
+
+	accessToken := "initial-access-token"
+
+	ts := startMockServer(t, &mockedAuthService, &mockedClient, accessToken, http.StatusOK, http.StatusUnauthorized)
+	defer ts.Close()
+
+	mockedAuthService.
+		On("Token").Return().
+		Twice()
+	mockedClient.
+		On("Test", "initial-access-token").Return().
+		Twice()
+
+	// When
+	c := client.NewWithSecret("my-client-id", "my-secret",
+		client.WithBaseURL(ts.URL)).HTTPClient()
+
+	// Then
+	resp, err := c.Get(ts.URL)
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusUnauthorized, resp.StatusCode)
+
+	mockedAuthService.AssertExpectations(t)
+	mockedClient.AssertExpectations(t)
+	mockedAuthService.AssertNotCalled(t, "Token", "refresh_token")
```

Comment:
I'm confused, shouldn't we assert that the token is refreshed?

Reply:
No, as we are not using refresh_token from OIDC protocol. In oauth2 s2c the refresh means "request a new one"